cdap-unit-test/src/test/java/io/cdap/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -1036,6 +1036,55 @@ public void testGetServiceURL() throws Exception {
     centralServiceManager.waitForStopped(10, TimeUnit.SECONDS);
   }
 
+  @Category(SlowTests.class)
+  @Test
+  public void testGetServiceURLDiffNamespace() throws Exception {
+    ApplicationManager discoveringApplicationManager = deployApplication(AppUsingGetServiceURL.class);
+    ServiceManager discoveringServiceManager = discoveringApplicationManager
+      .getServiceManager(AppUsingGetServiceURL.FORWARDING).start();
+    discoveringServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+    ApplicationManager systemApplicationManager = deployApplication(NamespaceId.SYSTEM, AppWithServices.class);
+    ServiceManager systemCentralServiceManager =
+      systemApplicationManager.getServiceManager(AppWithServices.SERVICE_NAME).start();
+    systemCentralServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
```

Comment:
please expand this test to cover a couple more scenarios:

 - Call the service before the system service is started
 - Call the service and pass it a non-existent namespace

and test the same except with the app in system namespace calling the app in the default namespace.

Reply:
Do you want me to expand the test or crate two new unit tests?

Reply:
expand this test because the overhead of these tests is very large.

Reply:
Interesting!

So, should I expand this test to include the two scenarios above and then, in the same test, test the, basically three scenarios? (i.e., the test would have 6 scenarios).

Reply:
yes but it should all be done in order, not just copy-pasting what would be multiple tests into one. In other words, it should:

start default app
call non-existent namespace from default
call system app from default 
start system namespace app
call system app from default
call default app from system
stop default app
call default app from system
stop system app